plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriter.java

```diff
@@ -113,7 +113,7 @@ public void write(CheckpointEntries entries, TrinoOutputFile outputFile)
         RowType metadataEntryType = checkpointSchemaManager.getMetadataEntryType();
         RowType protocolEntryType = checkpointSchemaManager.getProtocolEntryType(protocolEntry.getReaderFeatures().isPresent(), protocolEntry.getWriterFeatures().isPresent());
         RowType txnEntryType = checkpointSchemaManager.getTxnEntryType();
-        RowType addEntryType = checkpointSchemaManager.getAddEntryType(entries.getMetadataEntry(), writeStatsAsJson, writeStatsAsStruct);
+        RowType addEntryType = checkpointSchemaManager.getAddEntryType(entries.getMetadataEntry(), entries.getProtocolEntry(), writeStatsAsJson, writeStatsAsStruct);
```

Comment:
Given that your PR is equipped only with a test in `TestDeltaLakeAlterTableCompatibility` which exercises checkpoint writing for the situation when a table is written with writer version `7`, please add an extra test to ensure this code is properly covered in `TestDeltaLakeDatabricksCheckpointsCompatibility`.